### ISSUE TITLE
conventional-changelog-core version 5.0.1 throws `options.debug is not a function` 

### ISSUE THREAD

**Issue by ykageyama-mondo** [absolute-version/commit-and-tag-version#180]
**Describe the bug**

When the version of conventional-changelog-core resolves to `v5.0.1` due to package manager dependency resolution (e.g. installing this package after lerna [which pins to `v5.0.1`](https://github.com/lerna/lerna/blob/28c8ef232d29085ce9a9ba2bff96fae9017b3963/packages/lerna/package.json#L56)), running `npx commit-and-tag-version` causes `options.debug is not a function` to be thrown.

I believe this is likely caused by [this line](https://github.com/absolute-version/commit-and-tag-version/blob/a26d0482b18eff6dd5b22c183648ad2cd5e9352d/lib/lifecycles/changelog.js#L62) where the `debug` option is set to explicit `undefined` when `args.verbose` is not set. This causes downstream issues when [options is spread in conventional-changelog-core](https://github.com/conventional-changelog/conventional-changelog/blob/5b458dff9edf766446449acd1bbb8027a1e39fa5/packages/conventional-changelog-core/lib/merge-config.js#L88) which results in the default value of debug to be overridden by `undefined`. This behaviour is fixed in [conventional-changelog-core `v5.0.2`](https://github.com/conventional-changelog/conventional-changelog/releases/tag/conventional-changelog-core-v5.0.2)

**Current behavior**

Running `npx commit-and-tag-version` or `pnpm exec commit-and-tag-version` when `v5.0.1` is installed causes `Error in conventional-changelog-writer: options.debug is not a function` to be thrown.

[Minimal reproduction](https://github.com/ykageyama-mondo/conventional-changelog-version-issue-repro)

**Expected behavior**

Running `npx commit-and-tag-version`  or `pnpm exec commit-and-tag-version` should work.

**Environment**

- `commit-and-tag-version` version(s): v12.4.2
- Node/npm version: Node v18.17.1/pnpm 8.11.0
- OS: Ubuntu 22

**Possible Solutions**

1. Update to include a dependency for `conventional-changelog-core: "^5.0.2"`
2. Optionally add option keys instead of setting to undefined. e.g.
```ts
{
	//...
	...args.verbose && {debug: console.info.bind(console, 'conventional-changelog')},
	//...
}
```

**Additional context**

Occurs when commit-and-tag-version is installed into an existing package that already has `conventional-changelog-core@5.0.1` installed. Issue found when upgrading a lerna projen repo to [projen v0.86.2](https://github.com/projen/projen/releases/tag/v0.86.2)

**Comment by TimothyJones** [comment]
This should be fixed in 12.4.3.

**Comment by ykageyama-mondo** [comment]
> This should be fixed in 12.4.3.

Awesome! Thanks for resolving this

**Comment by TimothyJones** [comment]
No problem! Apologies for the delay
